privacyidea/lib/tokens/passwordtoken.py

```diff
@@ -64,7 +64,7 @@ def check_password(self, password):
             # getKey() returns bytes and since we can not assume, that the
             # password only contains printable characters, we need to compare
             # bytes strings here. This also avoids making another copy of 'key'.
-            if key == to_bytes(password):
+            if safe_compare(key, to_bytes(password)):
```

Comment:
1. `safe_compare` already does the conversion to bytes so this can be omitted here
2. We can call `self.secretObject.compare(password)` directly here, no need to decrypt the secret.

Reply:
I will change this an run the test.

Reply:
It looks like unfortunately we can not do this, since secretObject.compare assumes to compare a hexlified secret. Which the password is not. So we can not take the shortcut!

~~~~python
            res = -1
            if self.secretObject.compare(password):
                res = 0

            return res
~~~~

It would however not be an issue of the PR and we should not rewrite the password token on a bigger scale in a patch release.
So I ask you to merge this. Thx.